R/zchunk_L226.en_distribution_USA.R

```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
+             gas_adj =  if_else(State == "Hawaii", gcamusa.GAS_ADJ_THRESH,
+                                (Natural.gas - EIA_US_C_NG_DFO_prices$Natural.gas) * CONV_BTU_KJ * gdp_deflator(1975, 2009)),
+             liq_adj = (Distillate.fuel.oil - EIA_US_C_NG_DFO_prices$Distillate.fuel.oil)* CONV_BTU_KJ  * gdp_deflator(1975, 2009)) ->
+      EIA_tmp
+
+    # get maximum coal adjustment for replacing NA's:
+    EIA_tmp %>%
+      select(coal_adj) %>%
+      na.omit %>%
+      summarize(coal_adj = max(coal_adj)) %>%
+      as.double ->
+      maxCoalAdj
+
+    # use to replace NAs and finish adjustment calculations:
+    EIA_tmp%>%
+      replace_na(list(coal_adj = maxCoalAdj)) %>%
+      group_by(grid_region) %>%
+      summarize(coal_adj = median(coal_adj),
+                gas_adj = median(gas_adj),
+                liq_adj = median(liq_adj)) %>%
+      ungroup %>%
+      na.omit ->
+      L226.CostAdj_75USDGJ_FERC_F
+
+
+    # L226.TechCost_en_USA: cost adders
+    L226.TechShrwt_en_USA %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechYr"]])) %>%
+      mutate(minicam.non.energy.input = "regional price adjustment") %>%
+      left_join_error_no_match(L226.CostAdj_75USDGJ_FERC_F, by = c("region" = "grid_region")) %>%
+      rename(coal = coal_adj,
+             gas = gas_adj,
+             liquids = liq_adj) %>%
+      gather(sector1, adjustment, -region, -supplysector, -subsector, -technology, -year, -minicam.non.energy.input) %>%
+      mutate(tmp = supplysector) %>%
+      separate(tmp, c("trash1", "sector2", "trash2"), sep = " ") %>%
+      filter(sector1 == sector2) %>%
+      select(-trash1, -trash2, -sector1, -sector2) %>%
+      rename(input.cost = adjustment) %>%
+      mutate(input.cost = round(input.cost, gcamusa.DIGITS_COST)) ->
+      L226.TechCost_en_USA
+
+
+    # L226.Ccoef: carbon coef for cost adder sectors
+    L202.CarbonCoef %>%
+      filter(region == "USA") %>%
+      select(-region) ->
+      L226.Ccoef.usa
+
+    L226.TechCost_en_USA %>%
+      select(region, supplysector) %>%
+      distinct %>%
+      left_join_error_no_match(L226.Ccoef.usa, by = c("supplysector" = "PrimaryFuelCO2Coef.name")) ->
+      L226.Ccoef
+
+
+
+    # PART 2: ELECTRICITY TRANSMISSION AND DISTRIBUTION
+
+    # L226.DeleteSupplysector_USAelec: Removing the electricity T&D sectors of the USA region
+    # This should probably be converted to an assumption and read in at some point.
+    L226.DeleteSupplysector_USAelec <- tibble(region = "USA", supplysector = gcamusa.ELECT_TD_SECTORS)
+
+
+    ### Replacing for loop starting on line 152 in old DS. I /think/ these tibbles should be output in place of the "object" that currently is;
+    ### Otherwise just delete all of the below to line 250
+    ### There's also a couple inputs to this chunk that are NULL, and nothing gets done to: L226.SubsectorShrwt_en, L226.SubsectorInterpTo_en
+    L226.Supplysector_en %>%
```

Comment:
Your call but I'd just do these in one-liners, e.g. `L226.Supplysector_electd_USA <- global_energy_to_USA_electd(L226.Supplysector_en)`